### PR TITLE
python3Packages.annotated-types: init at 0.5.0

### PR DESCRIPTION
**Pull request by blaggacao**
###### Description of changes

Adds `python3Packages.annotated-types`: https://github.com/annotated-types/annotated-types

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by blaggacao** [comment]
Result of `nixpkgs-review pr 242542` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.annotated-types</li>
    <li>python311Packages.annotated-types</li>
  </ul>
</details>

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
the commit title contains an extra v prefix which must be stripped

File: pkgs/development/python-modules/annotated-types/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    repo = "annotated-types";
+    rev = "v${version}";
+    hash = "sha256-zCsWfJ8BQuov8FN+hlm9XBKWAAQ/KHPK/x024A8k2kE=";
+  };
+
+  nativeBuildInputs = [
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
```
Comment (natsukium) [inline pkgs/development/python-modules/annotated-types/default.nix:24]:
```suggestion
  propagatedBuildInputs = lib.optionals (pythonOlder "3.9") [
```
https://github.com/annotated-types/annotated-types/blob/0ad1bd8ea0323f270e5bb69f62235246ff80789e/pyproject.toml#L32

File: pkgs/development/python-modules/annotated-types/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    typing-extensions
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/annotated-types/default.nix:27]: Would you please add `pytestCheckHook`?

File: pkgs/development/python-modules/annotated-types/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  pythonImportsCheck = [ "annotated_types" ];
+
+  meta = with lib; {
+    description = "Reusable constraint types to use with typing.Annotated";
+    homepage = "https://github.com/annotated-types/annotated-types";
```
Comment (natsukium) [inline pkgs/development/python-modules/annotated-types/default.nix:32]:
```suggestion
    homepage = "https://github.com/annotated-types/annotated-types";
    changelog = "https://github.com/annotated-types/annotated-types/releases/tag/${src.rev}";
```
Adding the changelog to the meta is helpful for future updates.

File: pkgs/development/python-modules/annotated-types/default.nix (hunk 1 above)
Comment (blaggacao) [reply pkgs/development/python-modules/annotated-types/default.nix]: Thank you! Noted!

File: pkgs/development/python-modules/annotated-types/default.nix (hunk 3 above)
Comment (blaggacao) [reply pkgs/development/python-modules/annotated-types/default.nix:32]: Thanks, indeed!

File: pkgs/development/python-modules/annotated-types/default.nix (hunk 2 above)
Comment (blaggacao) [reply pkgs/development/python-modules/annotated-types/default.nix:27]: Beautiful:
image

**Comment by natsukium** [comment]
@ofborg build python310Packages.annotated-types

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 242542` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.annotated-types</li>
    <li>python310Packages.annotated-types.dist</li>
    <li>python311Packages.annotated-types</li>
    <li>python311Packages.annotated-types.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Pull request merged**